### ISSUE TITLE
error when zone missing SOA

### ISSUE THREAD

**Issue by digininja** [darkoperator/dnsrecon#49]
Looks like I messed up zonetransfer.me trying to get DNSSEC working but in the process broke dnsrecon...

```
./dnsrecon.py -t zonewalk -d zonetransfer.me -n 81.4.108.41
[*] Performing NSEC Zone Walk for zonetransfer.me
[*] Getting SOA record for zonetransfer.me
Traceback (most recent call last):
  File "./dnsrecon.py", line 1681, in <module>
    main()
  File "./dnsrecon.py", line 1595, in main
    ds_zone_walk(res, domain)
  File "./dnsrecon.py", line 1225, in ds_zone_walk
    soa_rcd = res.get_soa()[0][2]
  File "./recon/dnsrecon/lib/dnshelper.py", line 200, in get_soa
    ipv4_answers = self._res.query(name, 'A')
  File "/usr/lib/python2.7/dist-packages/dns/resolver.py", line 824, in query
    raise NoNameservers
dns.resolver.NoNameservers
```

**Comment by binarycanary** [comment]
Looks like a possible configuration issue with that name server returning REFUSED (response code 5) for queries. Not sure when, but it looks like the `/python2.7/site-packages/dns/resolver.py` now has a more verbose error message.

The final line of the traceback contains more info now:
**dns.resolver.NoNameservers: All nameservers failed to answer the query nsztm1.digi.ninja. IN A: Server 81.4.108.41 UDP port 53 answered REFUSED**

Could catch NoNameservers in `\lib\dnshelper.py`'s get_soa method and print out a nice error to the user before exiting. Not sure what else a good fix would be.

**Comment by GH0Z7S3C** [comment]
This probably didn't work because of a proxy service in use. I had similar problem until I turned off VPN. Check your proxy setup to see if this resolves issue.

**Comment by darkoperator** [comment]
Just come up with fix for this in 0.8.11, will push once I add other fixes

**Comment by mazen160** [comment]
Haven't seen this ticket, but seems related to my PR: https://github.com/darkoperator/dnsrecon/pull/83
It's now merged!
